features/nanostack/mbed-mesh-api/source/WisunInterface.cpp

```diff
@@ -329,6 +329,34 @@ mesh_error_t WisunInterface::set_network_phy_mode_and_channel_plan_id(uint8_t ph
     return MESH_ERROR_NONE;
 }
 
+mesh_error_t WisunInterface::get_network_phy_mode_and_channel_plan_id(uint8_t *phy_mode_id, uint8_t *channel_plan_id)
+{
+    int status = ws_management_phy_mode_id_get(get_interface_id(), phy_mode_id);
+    if (status != 0) {
+        return MESH_ERROR_UNKNOWN;
+    }
+    status = ws_management_channel_plan_id_get(get_interface_id(), channel_plan_id);
+    if (status != 0) {
+        return MESH_ERROR_UNKNOWN;
+    }
+
+    return MESH_ERROR_NONE;
+}
+
+mesh_error_t WisunInterface::validate_network_phy_mode_and_channel_plan_id(uint8_t phy_mode_id, uint8_t channel_plan_id)
+{
+    int status = ws_management_phy_mode_id_validate(get_interface_id(), phy_mode_id);
```

Comment:
Did we make separate validates for mode and plan id. this sounds problematic as there is probably some mode, plan combinations we don't support.
supported 
mode1,plan1
mode2,plan1
mode2,plan2
but not supported
mode2,plan2
we can't set or validate this proberly do we need to change the underlying API?

Reply:
I see your point. Yes, the validation on Nanostack API is separated. We can keep this PR on hold and I can change the Nanostack API.